### PR TITLE
domain: fast new a etcd session when the session is stale in the schemaVersionSyncer

### PR DESCRIPTION
**Pull request by winkyao**
Fix issue https://internal.pingcap.net/jira/browse/TIDB-1668

I have done a manual test using ansible, with inventory.ini:

```
## TiDB Cluster Part
[tidb_servers]
192.168.59.105

[tikv_servers]
192.168.59.105

[pd_servers]
192.168.59.105
192.168.59.103

[spark_master]

[spark_slaves]

## Monitoring Part
# prometheus and pushgateway servers
[monitoring_servers]
192.168.59.105

[grafana_servers]
192.168.59.105

# node_exporter and blackbox_exporter servers
[monitored_servers]
192.168.59.105

[alertmanager_servers]
192.168.59.105

[kafka_exporter_servers]

## Binlog Part
[pump_servers:children]
tidb_servers

[drainer_servers]

## Group variables
[pd_servers:vars]
# location_labels = ["zone","rack","host"]

## Global variables
[all:vars]
deploy_dir = /home/tidb/deploy

## Connection
# ssh via normal user
ansible_user = wink

cluster_name = test-cluster

tidb_version = v2.0.1

# process supervision, [systemd, supervise]
process_supervision = systemd

# timezone of deployment region
timezone = Asia/Shanghai
set_timezone = True

enable_firewalld = False
# check NTP service
enable_ntpd = True
set_hostname = False
# CPU, memory and disk performance will not be checked when dev_mode = True
dev_mode = False

## binlog trigger
enable_binlog = False
# zookeeper address of kafka cluster for binlog, example:
# zookeeper_addrs = "192.168.0.11:2181,192.168.0.12:2181,192.168.0.13:2181"
zookeeper_addrs = ""
# kafka cluster address for monitoring, example:
# kafka_addrs = "192.168.0.11:9092,192.168.0.12:9092,192.168.0.13:9092"
kafka_addrs = ""

# store slow query log into seperate file
enable_slow_query_log = False

# enable TLS authentication in the TiDB cluster
enable_tls = False

# KV mode
deploy_without_tidb = False

# Optional: Set if you already have a alertmanager server.
# Format: alertmanager_host:alertmanager_port
alertmanager_target = ""

grafana_admin_user = "admin"
grafana_admin_password = "admin"


### Collect diagnosis
collect_log_recent_hours = 2

enable_bandwidth_limit = True
# default: 10Mb/s, unit: Kbit/s
collect_bandwidth_limit = 10000
```
There are 2 pd, 1 tidb, 1 tikv in the test cluster, I start a client which continue inserting data into tidb, and I kill -9 the pd leader, we can see the tidb report errors:

```
2018/09/28 11:53:58.800 domain.go:417: [info] [ddl] reload schema in loop, server info syncer need restart
```

After the pd recover: `2018/09/28 11:54:59.562 leader.go:269: [info] PD cluster leader pd1 is ready to serve`, the etcd session is recovered:

```
2018/09/28 11:54:59.545 domain.go:419: [info] [ddl] server info syncer restarted.
```
It costs several millseconds to recover.

### CODE REVIEW

File: ddl/syncer.go [hunk 1]
```diff
@@ -88,7 +90,7 @@ type SchemaSyncer interface {
 type schemaVersionSyncer struct {
 	selfSchemaVerPath string
 	etcdCli           *clientv3.Client
-	session           *concurrency.Session
+	session           unsafe.Pointer
```
[the remark this answers is not in the document]
Comment (winkyao) [reply ddl/syncer.go]: session is accessed in multi-thread.

**Comment by winkyao** [comment]
@zimulala @crazycs520 @shenli PTAL

**Comment by ciscoxll** [comment]
@zimulala PTAL.

File: domain/domain.go [hunk 2]
```diff
@@ -405,16 +407,16 @@ func (do *Domain) loadSchemaInLoop(lease time.Duration) {
 		case <-syncer.Done():
 			// The schema syncer stops, we need stop the schema validator to synchronize the schema version.
 			log.Info("[ddl] reload schema in loop, schema syncer need restart")
-			do.SchemaValidator.Stop()
 			err := do.mustRestartSyncer()
 			if err != nil {
 				log.Errorf("[ddl] reload schema in loop, schema syncer restart err %v", errors.ErrorStack(err))
 				break
 			}
-			do.SchemaValidator.Restart()
```
Comment (zimulala) [inline domain/domain.go]: Need to add metrics here?

Comment (winkyao) [reply domain/domain.go]: we already has metrics.NewSessionHistogram and metrics.DeploySyncerHistogram, I think there is nothing need to add here.

Comment (zimulala) [reply domain/domain.go]: Got it.

File: ddl/syncer.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (zimulala) [reply ddl/syncer.go]: Can you talk about it in detail?

[the remark this answers is not in the document]
Comment (zimulala) [reply ddl/syncer.go]: Gotcha.

**Pull request merged**